### PR TITLE
Added event processor metrics

### PR DESCRIPTION
**Pull request by duke8253**

### CODE REVIEW

**Comment by zwoop** [comment]
Ok, I do like the style of this PR, this is good. Question: Does this have an impact on performance at all? Is there a noticeable amount of computational in the event loop now?

**Comment by zwoop** [comment]
[approve ci].

**Comment by zwoop** [comment]
Shouldn't we encourage adding these new metrics to the documentation too ?

**Comment by duke8253** [comment]
The computation should be minimal in the added metrics, it's basically just counting and records timestamps.

**Comment by SolidWallOfCode** [comment]
I should point out, to embarrass @zwoop, that I originally wrote this code, @duke8253 is porting it forward to 7.1.x from our internal 5.3.x. I designed it to have as small a footprint as possible, but we should probably do some tests on its impact. It's one reason I wanted to separate it out from the precursor event loop changes. I know @duke8253 loves to do performance tests, maybe he'll help out with that?

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Comment by SolidWallOfCode** [comment]
Also, @zwoop is correct, this should not be approved until there is documentation for the metrics. I'll try to work on that.

**Comment by shinrich** [comment]
Looks like autest failed due to the new txn tests.  @lbordowitz could you take a look?

File: iocore/eventsystem/UnixEThread.cc [hunk 1]
```diff
@@ -201,6 +239,7 @@ EThread::execute()
             NegativeQueue.insert(e, p);
           }
         }
+        ++nq_count;
```
Comment (oknet) [inline iocore/eventsystem/UnixEThread.cc]: Should be place after `NegativeQueue.insert(e, p);` ?

Comment (oknet) [reply iocore/eventsystem/UnixEThread.cc]: There is no one read the `nq_count`.

File: iocore/eventsystem/UnixEThread.cc [hunk 2]
```diff
@@ -265,12 +304,15 @@ EThread::execute()
         while ((e = NegativeQueue.dequeue())) {
           process_event(e, EVENT_POLL);
         }
+
+        metric_wait_check = EventQueue.earliest_timeout() - cur_time;
+
```
Comment (oknet) [inline iocore/eventsystem/UnixEThread.cc]: There is no sleep_time if NegativeQueue is not empty.
```
EventQueueExternal.dequeue_timed(cur_time, next_time, false);
```
The 3rd parameter is `false` means no condition wait and return immediately.

**Comment by SolidWallOfCode** [comment]
The original code was written for the changes in #2541 and are a bit off due to the back port.

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Pull request merged**